warehouse/query-core/src/main/java/datawave/next/scanner/KeyWithContext.java

```diff
@@ -0,0 +1,58 @@
+package datawave.next.scanner;
+
+import org.apache.accumulo.core.data.Key;
+
+import datawave.core.query.configuration.QueryData;
+
+/**
+ * Simple object that allows state to travel with a document id
+ */
+public class KeyWithContext implements Comparable<KeyWithContext> {
+
+    private final Key key;
+    private final QueryData context;
+    private final String compareKey;
+
+    /**
+     * Default constructor
+     *
+     * @param key
+     *            the record id
+     * @param context
+     *            the context
+     */
+    public KeyWithContext(Key key, QueryData context) {
+        this(key, context, false);
+    }
+
+    public KeyWithContext(Key key, QueryData context, boolean buildCompareKey) {
+        this.key = key;
+        this.context = context;
+
+        if (buildCompareKey) {
+            // the ColumnFamily is datatype \0 uid, invert this for priority queue
+            String cf = key.getColumnFamily().toString();
+            int index = cf.indexOf('\u0000');
+            compareKey = cf.substring(index + 1) + cf.substring(0, index);
```

Comment:
this is dropping the null byte from the key, is this desirable?

Reply:
Unintended, but doesn't have any bad side effects